moabb/evaluations/splitters.py

```diff
@@ -101,3 +110,165 @@ def split(self, y, metadata):
+
+    It assumes that the entire metainformation across all subjects is already loaded.
+
+    Unlike the `CrossSessionEvaluation` class from `moabb.evaluation`, which manages
+    the complete evaluation process end-to-end, this splitter is solely responsible
+    for dividing the data into training and testing sets based on sessions.
+
+    .. image:: images/crosssess.jpg
+        :alt: The schematic diagram of the CrossSession split
+        :align: center
+
+    The inner cross-validation strategy can be changed by passing the
+    `cv_class` and `cv_kwargs` arguments. By default, it uses LeaveOneGroupOut,
+    which effectively performs Leave-One-Session-Out cross-validation when
+    sessions are passed as groups.
+
+    Parameters
+    ----------
+    cv_class: cross-validation class, default=LeaveOneGroupOut
+        Inner cross-validation strategy for splitting the sessions.
+        For cross-session splitting, LeaveOneGroupOut is the most suitable as default.
+    shuffle: bool, default=False
+        Whether to shuffle the session order for each subject. By default, it is not
+        used because of LeaveOneGroupOut's determinist behaviour.
+    random_state: int, RandomState instance or None, default=None
+        Controls the randomness when `shuffle` is True.
+        Pass an int for reproducible output across multiple function calls.
+        Be default, it is not used because of the determinist behaviour of
+        LeaveOneGroupOut.
+    cv_kwargs: dict
+        Additional arguments to pass to the inner cross-validation strategy.
+
+    Yields
+    ------
+    train : ndarray
+        The training set indices for that split.
+
+    test : ndarray
+        The testing set indices for that split.
+    """
+
+    def __init__(
+        self,
+        cv_class: type[BaseCrossValidator] = LeaveOneGroupOut,
+        shuffle: bool = False,
+        random_state: int = None,
+        **cv_kwargs,
+    ):
+        self.cv_class = cv_class
+        self._cv_kwargs = cv_kwargs
+
+        self.shuffle = shuffle
+        self.random_state = random_state
+
+        if not shuffle and random_state is not None:
+            raise ValueError("`random_state` should be None when `shuffle` is False")
+
+        if shuffle and len(self._cv_kwargs) == 0 and cv_class is LeaveOneGroupOut:
+            raise ValueError(
+                "Shuffling is not implemented for LeaveOneGroupOut. "
+                "The `shuffle` parameter change the behaviour of the splitter."
+                "Use GroupShuffleSplit instead."
+            )
+
+    @property
+    def cv_kwargs(self):
+        params = inspect.signature(self.cv_class).parameters
+        rng = None
+        if self.shuffle:
+            rng = check_random_state(self.random_state)
+        for p, v in [
+            ("shuffle", self.shuffle),
+            ("random_state", rng),
+        ]:
+            if p in params:
+                self._cv_kwargs[p] = v
```

Comment:
I don't like when a property is doing some computation. It should only be an accessor, not check the signature of a function. I would rather keep the original design.

it is also more in line with the sklearn API, which is to store all parameters as is in the constructor and store privately the updated values.

Reply:
Also, here the issue is that you will have the same seed for all subjects.